app/invocation/invocation_build_logs_card.tsx

```diff
@@ -1,89 +1,30 @@
-import InvocationModel from "./invocation_model";
 
 interface Props {
-  model: InvocationModel;
+  value: string;
+  loading: boolean;
   expanded: boolean;
   dark: boolean;
 }
 
-interface State {
-  consoleBuffer?: string;
-}
-
-const POLL_TAIL_INTERVAL_MS = 3_000;
-// How many lines to request from the server on each chunk request.
-const MIN_LINES = 100_000;
-
-export default class BuildLogsCardComponent extends React.Component<Props, State> {
-  state: State = {
-    consoleBuffer: "",
-  };
-
-  private pollTailTimeout: number | null = null;
-
-  componentDidMount() {
-    if (this.props.model.hasChunkedEventLogs()) {
-      this.fetchTail();
-    }
-  }
-
-  componentWillUnmount() {
-    window.clearTimeout(this.pollTailTimeout);
-  }
-
-  private fetchTail(chunkId = "") {
-    rpcService.service
-      .getEventLogChunk(
-        new eventlog.GetEventLogChunkRequest({
-          invocationId: this.props.model.getId(),
-          chunkId,
-          minLines: MIN_LINES,
-        })
-      )
-      .then((response) => {
-        this.setState({
-          consoleBuffer: this.state.consoleBuffer + String.fromCharCode(...(response.buffer || [])),
-        });
-
-        // Empty next chunk ID means the invocation is complete and we've reached
-        // the end of the log.
-        if (!response.nextChunkId) return;
-
-        // Unchanged next chunk ID means the invocation is still in progress and
-        // we should continue polling that chunk.
-        if (response.nextChunkId === chunkId) {
-          this.pollTailTimeout = window.setTimeout(() => this.fetchTail(chunkId), POLL_TAIL_INTERVAL_MS);
-          return;
-        }
-
-        // New next chunk ID means we successfully fetched the requested
-        // chunk, and more may be available. Try fetching it immediately.
-        this.fetchTail(response.nextChunkId);
-      })
-      .catch((e) => errorService.handleError(e));
-  }
-
-  private getConsoleBuffer() {
-    if (!this.props.model.hasChunkedEventLogs()) {
-      return this.props.model.consoleBuffer || "";
-    }
-
-    return this.state.consoleBuffer;
-  }
-
+export default class BuildLogsCardComponent extends React.Component<Props> {
   render() {
     return (
       <div className={`card ${this.props.dark ? "dark" : "light-terminal"} ${this.props.expanded ? "expanded" : ""}`}>
         <img className="icon" src={this.props.dark ? "/image/log-circle-light.svg" : "/image/log-circle.svg"} />
         <div className="content">
           <div className="title">Build logs </div>
           <div className="details">
-            <TerminalComponent value={this.getConsoleBuffer()} lightTheme={!this.props.dark} />
+            {this.props.loading ? (
+              // "terminal" class makes sure the loading container's size
+              // is the same as the terminal that will take its place.
+              <div className="terminal">
+                <div className="loading" />
```

Comment:
does this handle dark and light terminal well?

Reply:
good catch - fixed

![image](https://user-images.githubusercontent.com/2414826/127720043-34c2c3b5-68e7-45bb-a503-ba86a64b6ff8.png)